core/src/main/java/org/commonjava/maven/ext/core/impl/DependencyManipulator.java

```diff
@@ -390,12 +391,13 @@ private void apply( final Project project, final Model model, final Map<Artifact
             applyExplicitOverrides( project, project.getResolvedManagedDependencies( session ), explicitOverrides,
                                     commonState, explicitVersionPropertyUpdateMap );
 
-            if ( commonState.isOverrideTransitive() )
+            if ( commonState.isOverrideTransitive() && dependencyState.getRemoteBOMDepMgmt() != null )
             {
-                final List<Dependency> extraDeps = new ArrayList<>();
+                final Collection<ArtifactRef> overrideRefs = overrides.keySet();
+                final Collection<Dependency> extraDeps = new ArrayList<>( overrideRefs.size() );
 
                 // Add dependencies to Dependency Management which did not match any existing dependency
-                for ( final ArtifactRef pvr : overrides.keySet() )
+                for ( final ArtifactRef pvr : overrideRefs )
```

Comment:
Why can't overridesRefs be inlined to `overrides.keySet()` ?

Reply:
I pulled `overrideRefs = overrides.keySet()` out of the loop in order to make a small optimization from

```
final List<Dependency> extraDeps = new ArrayList<>();
```

to

```
final Collection<Dependency> extraDeps = new ArrayList<>( overrideRefs.size() );
```

since we know it's (maximum) size ahead of time can only be equal to the size of `overrides`.

Reply:
Default capacity is 10 I believe and it increases in O(n) time. So whether in reality this causes an actual improvement is doubtful. However, its harmless enough so we could keep it.

Reply:
True, this list in particular is probably so small as to not matter. I imagine the majority of overrides will fall under 10 anyway, or at least between 10 and 20.

I guess it's more of a question of what the general policy should be. My feeling is that if we know a list's size ahead of time, we should definitely put the size. If we just know its upper bound, I guess that's more debatable. This bound means that it won't have to grow, which I understand is the more expensive operation vs. using a little bit of extra memory at the outset.

Reply:
Agreed, as a general policy I guess if we have the information we can use it.